src/library_browser.js

```diff
@@ -832,27 +832,49 @@ var LibraryBrowser = {
   emscripten_async_run_script__deps: ['emscripten_run_script', '$safeSetTimeout'],
   emscripten_async_run_script: function(script, millis) {
     // TODO: cache these to avoid generating garbage
-    safeSetTimeout(function() {
-      _emscripten_run_script(script);
-    }, millis);
+    safeSetTimeout(() => _emscripten_run_script(script), millis);
   },
 
   // TODO: currently not callable from a pthread, but immediately calls onerror() if not on main thread.
+  emscripten_async_load_script__deps: ['$UTF8ToString'],
   emscripten_async_load_script: function(url, onload, onerror) {
+    url = UTF8ToString(url);
     onload = {{{ makeDynCall('v', 'onload') }}};
     onerror = {{{ makeDynCall('v', 'onerror') }}};
 
 #if PTHREADS
     if (ENVIRONMENT_IS_PTHREAD) {
-      err('emscripten_async_load_script("' + UTF8ToString(url) + '") failed, emscripten_async_load_script is currently not available in pthreads!');
+      err('emscripten_async_load_script("' + url + '") failed, emscripten_async_load_script is currently not available in pthreads!');
       return onerror ? onerror() : undefined;
     }
 #endif
+    assert(runDependencies === 0, 'async_load_script must be run when no other dependencies are active');
+
     {{{ runtimeKeepalivePush() }}}
 
-    assert(runDependencies === 0, 'async_load_script must be run when no other dependencies are active');
+#if ENVIRONMENT_MAY_BE_NODE
+    if (ENVIRONMENT_IS_NODE) {
+      readAsync(url, (data) => {
```

Comment:
It looks like the functions here can be shared with below? The main difference is just in calling `readAsync` vs setting them on `.onload/onerror`.